server/publish.js

```diff
@@ -1,70 +1,85 @@
 Meteor.publish("chatterMessages", function (opts) {
-    check(opts, {
-        roomId: String,
-        messageLimit: Number
-    });
+  check(opts, {
+    roomId: String,
+    messageLimit: Number
+  });
 
-    /*
-    let room = ChatterRoom.findOne({ name: opts.roomName });
-    if (_.isUndefined(room)) return [];
-    // TODO Check for room restrictions here
-    */
+  /*
+  let room = ChatterRoom.findOne({ name: opts.roomName });
+  if (_.isUndefined(room)) return [];
+  // TODO Check for room restrictions here
+  */
 
-    return ChatterMessage.find({
-        roomId: opts.roomId,
-    }, {
-        limit: opts.limit,
-        fields: {
-            message: 1,
-            roomId: 1,
-            userNick: 1,
-            userAvatar: 1,
-            userId: 1,
-            createdAt: 1
-        }
-    });
+  return ChatterMessage.find({
+    roomId: opts.roomId,
+  }, {
+    limit: opts.limit,
+    fields: {
+      message: 1,
+      roomId: 1,
+      userNick: 1,
+      userAvatar: 1,
+      userId: 1,
+      createdAt: 1
+    }
+  });
 });
 
 Meteor.publish("chatterRooms", function () {
-    let user = Meteor.users.findOne(this.userId);
+  let user = Meteor.users.findOne(this.userId);
 
-    // TODO do some clever things here to determine
-    // which rooms are accessible
+  // TODO do some clever things here to determine
+  // which rooms are accessible
 
-    return ChatterRoom.find({
-    }, {
-        fields: {
-            name: 1,
-            roomType: 1,
-            lastActive: 1
-        }
-    });
+  return ChatterRoom.find({
+  }, {
+    fields: {
+      name: 1,
+      description: 1,
+      roomType: 1,
+      lastActive: 1,
+      archived: 1,
+      createdAt: 1
+    }
+  });
 });
 
 Meteor.publish("chatterUserRooms", function () {
-    let user = Meteor.users.findOne(this.userId);
+  let user = Meteor.users.findOne(this.userId);
 
-    // TODO do some clever things here to determine
-    // which rooms are accessible
+  // TODO do some clever things here to determine
+  // which rooms are accessible
 
-    return ChatterUserRoom.find({
-    }, {
-        fields: {
-            count: 1,
-            userId: 1,
-            roomId: 1
-        }
-    });
+  return ChatterUserRoom.find({
+  }, {
+    fields: {
+      count: 1,
+      userId: 1,
+      roomId: 1
+    }
+  });
 });
 
-Meteor.publish("chatterUsers", function (roomId) {
-    // TODO do some clever things here to determine
-    // which users are visible
-    return Meteor.users.find({
-        "status.online": true
-    }, {
-        userNick: 1,
-        avatar: 1
-    });
+Meteor.publish("users", function (roomId) {
+  // TODO do some clever things here to determine
+  // which users are visible
+  return Meteor.users.find({
+    "status.online": true
+  }, {
+    userNick: 1,
```

Comment:
The limit on publish is there to make sure that you don't load more documents than necessary.
I don't see what you mean regarding the Meteor call, why would a room creation method take a message limit? Was that something I wrote at some point?
I see no reason why such a message limit cannot be set on the server, no, and it probably should. However, a few times when you sub to a dataset, you only want the latest 10 for example. It makes sense then to allow a limit parameter in the pub/sub call (temporary).

With the last question, do you mean the messages that are "over" the limit, so to say? If you don't have a `sort` in the publish query (like by createdAt for exampe), you will get a semi-random sample of the limit number of elements, the rest will just not get sent, so the user won't get them. Usually in this situation you would sort the collection to send the most recent documents first, and just don't care about the rest. Usually this is fine, but say you wanted to see the whole collection of some thousands of documents. Then yes, you'd have to paginate in some way. Many packages for this on atmosphere luckily!